### ISSUE TITLE
Add Click to copy text for commands present in the official Django website.

### ISSUE THREAD

**Issue by VenkataBhaskarr** [django/djangoproject.com#1276]
As I am following through the Django project tutorial in the official Django website there are so many commands like below, so adding a click to copy text would be a time saver.

Before : 

<img width="783" alt="Screenshot 2022-11-23 at 11 32 30 AM" src="https://user-images.githubusercontent.com/111889155/203481536-369be822-7f8d-479d-b151-cf118f57616e.png">

After : 

<img width="704" alt="Screenshot 2022-11-23 at 9 30 01 PM" src="https://user-images.githubusercontent.com/111889155/203592385-419d0311-3a58-4ca0-8ff7-26e69544dd20.png">

**Comment by himanshu007-creator** [comment]
I would like to work on that issue

**Comment by carltongibson** [comment]
There's a Sphinx copy button extension: https://sphinx-copybutton.readthedocs.io/en/latest/ — looking into adapting that would be good. 

* What's the HTML it outputs? 
* What's the static file we'd need to include. 

The docs aren't raw sphinx HTML. We use the JSON builder, and then insert the body into a Django template.

**Comment by VenkataBhaskarr** [comment]
yes if we used Sphinx copy button extension the ouputs are as follows 

I have added ```sphinx_copybutton``` in extensions in conf.py file in docs folder.


BEFORE :

<img width="668" alt="Screenshot 2022-11-26 at 10 56 32 AM" src="https://user-images.githubusercontent.com/111889155/204073990-4ba46b74-acb5-4d20-bc55-02d1714a5739.png">


AFTER : 


<img width="663" alt="Screenshot 2022-11-26 at 10 56 24 AM" src="https://user-images.githubusercontent.com/111889155/204073993-c45c787b-12fc-4580-baaa-197df9fa20e3.png">

**Comment by Maniktherana** [comment]
It seems some code blocks already have this feature. Perhaps we can just utilize that but I like the style of the Sphinx copy button for 1 line commands.

 I'd be willing to work on this if it's still open.

<img width="929" alt="Screenshot 2023-01-23 at 4 08 48 PM" src="https://user-images.githubusercontent.com/14011425/214019444-e8f36ac7-3ebf-478b-9257-180d35f5c463.png">

**Comment by GitAditya123** [comment]
>>> import django
>>> print(django.get_version())
4.2

**Comment by benjaoming** [comment]
sphinx-copybutton also uses clipboard.js - so I guess it's a matter of avoiding loading the same JS library twice if its included through the Sphinx extension. If the copy of the JS library included in `sphinx-copybutton` can be used for the Django project (or the other way around), then we can also avoid having the same asset from different sources and with different versions. But other than that, seems like it's possible to use the same styles, HTML elements and JS library on the entire website, as well as isolated Sphinx documentation builds :+1:

**Comment by rabahalishah** [comment]
I would like to contribute to this.

**Comment by sabderemane** [comment]
@rabahalishah feel free to work on this, no need to be assigned to work on an issue :)

**Comment by marksweb** [comment]
There's a class of `code-block-caption` which adds the styling and the copy button, but the JS doesn't appear to be loaded by default.

This class and `snippet` are hooked up to `clippify`, but they don't appear to be in use anywhere at the moment.

https://github.com/django/djangoproject.com/blob/main/djangoproject/static/js/main.js#L101

**Comment by Yadnyesh-Dashpute** [comment]
@VenkataBhaskarr  Please Assign Me This Issue. I Want To Work On this Issue.

**Comment by marksweb** [comment]
@Yadnyesh-Dashpute you don't need to be assigned. People can see now that you will be working on this so go ahead and raise a PR when you have something to look at 👍

**Comment by Yadnyesh-Dashpute** [comment]
@marksweb Okay

**Comment by Om15102003** [comment]
I would like to work on this issue.

**Comment by marksweb** [comment]
@Om15102003 There's already an open PR for this here; #1434 

If you find something you'd like to work on, just go ahead and work on it - there's no need to ask. Though it's worth looking to see if anybody has mentioned that they might be working on it or if there are linked PRs. Quite often someone might say they're going to look at something, but unless there's a PR we don't really know the status of that, so bare in mind how long it has been since someone talked about working on something.

**Comment by Fareed95** [comment]
Hello sir. I'm  very new in opensource coding.. but I think I can resolve this problem . but I'm confused from which file is this issue is related to ?? would you please guide me ??

**Comment by varshithreddy7** [comment]
I will like to contribute this issue

**Comment by 08preeti** [comment]
The code provided is for adding a [Click to copy]functionality. When users click the "Click to copy" button next to a command or piece of text, the text is automatically copied to their clipboard.

<!DOCTYPE html>
<html lang="en">
<head>
    <meta charset="UTF-8">
    <meta name="viewport" content="width=device-width, initial-scale=1.0">
    <title>Click to Copy Example</title>
    <style>
        .command-container {
            position: relative;
            display: inline-block;
            margin-top: 10px;
        }
        .copy-button {
            position: absolute;
            top: 0;
            right: 0;
            padding: 5px;
            cursor: pointer;
            background-color: #f3f3f3;
            border: 1px solid #ccc;
            border-radius: 3px;
        }
    </style>
</head>
<body>

<h3>Example Django Command</h3>

<div class="command-container">
    <pre><code id="commandText">python manage.py migrate</code></pre>
    <button class="copy-button" onclick="copyToClipboard('commandText')">Click to copy</button>
</div>

<script>
    function copyToClipboard(elementId) {
        const textToCopy = document.getElementById(elementId).innerText;
        const tempTextArea = document.createElement("textarea");
        tempTextArea.value = textToCopy;
        document.body.appendChild(tempTextArea);
        tempTextArea.select();
        document.execCommand("copy");
        document.body.removeChild(tempTextArea);
        alert("Command copied to clipboard!");
    }
</script>

</body>
</html>

**Comment by parthhhgohel** [comment]
I have solution of this issue

**Comment by divya5-11-04** [comment]
<!DOCTYPE html>
<html lang="en">
<head>
    <meta charset="UTF-8">
    <meta name="viewport" content="width=device-width, initial-scale=1.0">
    <title>Click to Copy Example</title>
    <style>
        .command-box {
            display: flex;
            align-items: center;
            gap: 10px;
            background-color: #f4f4f4;
            border: 1px solid #ccc;
            padding: 10px;
            border-radius: 5px;
            font-family: monospace;
        }
        .copy-btn {
            background-color: #007acc;
            color: white;
            border: none;
            border-radius: 3px;
            padding: 5px 10px;
            cursor: pointer;
        }
        .copy-btn:hover {
            background-color: #005f99;
        }
    </style>
</head>
<body>
    <div class="command-box">
        <span>python manage.py runserver</span>
        <button class="copy-btn" onclick="copyText(this)">Copy</button>
    </div>

    <div class="command-box">
        <span>django-admin startproject myproject</span>
        <button class="copy-btn" onclick="copyText(this)">Copy</button>
    </div>

    <script>
        function copyText(button) {
            const command = button.previousElementSibling.textContent;
            navigator.clipboard.writeText(command)
                .then(() => {
                    button.textContent = "Copied!";
                    setTimeout(() => (button.textContent = "Copy"), 2000);
                })
                .catch(err => {
                    console.error("Error copying text: ", err);
                });
        }
    </script>
</body>
</html>
 This code snippets can be integrated in documentation where commands are given

**Comment by VidhiKathiriya05** [comment]
Hi @VenkataBhaskarr 
I’d be happy to work on this issue.

As someone who uses the Django documentation a lot, I completely understand how helpful this improvement would be. Instead of adding a simple copy button, I’d like to propose a more helpful “Smart Copy” feature that improves the user experience.

My Plan
I. Smart Copy Logic (Better UX)

One of the most common problems in technical docs is that users accidentally copy the shell prompts ($ or >>>) along with the command.

How it will work:
When the user clicks the copy button, the script will check the code block using regex.

What will happen:
The script will automatically remove any leading prompts like $, %, or >>> before copying the text.

Example:
Shown in docs:
$ python -m pip install Django

Copied to clipboard:
python -m pip install Django

II. Technical Approach (Lightweight & Clean)

Stack: Plain JavaScript (no jQuery or external libraries), keeping it fast and lightweight.
Clipboard API: Uses navigator.clipboard.writeText() with a fallback if needed.
Integration: This will be added through static JS in the Sphinx theme so it works across all code blocks (e.g., with the .highlight class).

CODE LOGIC EXAMPLE:
const textToCopy = codeBlock.innerText;
// Remove shell prompts like $ or >>> at the start of lines
const cleanText = textToCopy.replace(/^(?:\$|>>>)\s/gm, '');
navigator.clipboard.writeText(cleanText);


I’ve worked with Sphinx themes and vanilla JavaScript before, so I can prepare a PR within a couple of days.

Let me know if this direction matches what the maintainers have in mind!

**Comment by adamzap** [comment]
Closed by #2431.
